authority/ssh.go

```diff
@@ -199,6 +205,26 @@ func (a *Authority) SignSSH(ctx context.Context, key ssh.PublicKey, opts provisi
 		Key:        key,
 	}
 
+	// Call enriching webhooks
+	if webhookCtl != nil {
+		whEnrichReq, err := webhook.NewRequestBody(
+			webhook.WithSSHCertificateRequest(cr),
+		)
+		if err != nil {
+			return nil, errs.ApplyOptions(
+				errs.InternalServerErr(err),
+				errs.WithKeyVal("signOptions", signOpts),
+			)
+		}
+		err = webhookCtl.Enrich(whEnrichReq)
```

Comment:
This doesn't look like the natural place for this call. The template data is always defined in `CustomSSHTemplateOptions` (or `CustomTemplateOptions` for X509) in the provisioner package.

You have managed to solve it with a setter. But I think in a previous commit you had that embedded in the above methods, which feels more natural. It was a little complex because there were too many composing functions in the same function, but it was easy to solve by extracting that code to a new method.

Reply:
I think it works well here now that all enriching webhooks are also authorizing webhooks. I think the most common error will be the webhook does not allow the request to continue. It will also be easier to add in authority-level webhooks in the future.